src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/PreparedStatementTest.java

```diff
@@ -295,6 +296,183 @@ public void testStatementPooling() throws SQLException {
         } 
     }
 
+    /**
+     * Test handling of statement pooling for prepared statements.
+     * 
+     * @throws SQLException
+     * @throws SecurityException 
+     * @throws NoSuchFieldException 
+     * @throws IllegalAccessException 
+     * @throws IllegalArgumentException 
+     */
+    @Test
+    @Tag("slow")
+    public void testStatementPoolingUseBulkCopyAPI() throws SQLException, NoSuchFieldException, SecurityException, 
```

Comment:
A lot of duplication here. Can't we refactor this?

Reply:
Done.